### DISCUSSION TITLE
The compass function is abnormal

### DISCUSSION THREAD

**Discussion by dizzyelk** [Zrips/CMI#7217] [Enhancement]
As for the Compass function of CMI, show Compass Target must show Spawn, Is it a bug?

**Comment by mrfloris** [comment]
Can you describe this a bit better if setting it to false still shows it or something?

Perhaps include the full output of
```
/ver
/cmi version
```
I see config.yml from cmi has these settings:
```yaml
  ShowSpawn: true
  Spawn: ⤋
```
If you do not wish to show the spawn you can set this to false. 

<img width="1211" alt="Screen-Shot 2022-09-25 at 17 32 02" src="https://user-images.githubusercontent.com/28841349/192151759-f0fe45f1-6f18-4e22-bedd-08e6db60b721.png">

**Reply by dizzyelk** [reply]
I want to display CompassTarget, but I don't want to display Spawn. Therefore, when I set ShowSpawn to false and ShowCompassTarget to true, CompassTarget will not be displayed in Compass because it is affected by ShowSpawn.

![image](https://user-images.githubusercontent.com/43479271/192438700-d2464483-4793-4712-b40c-5474f7205fcb.png)

**Comment by CloudeLecaw** [comment]
u may used this command? If Not it means no target was set. In case no Target was set it means that the spawn is set as target.
and if you hide spawn it means default target shows nothing...that simple..
If you want a fixed target for all i recomment using the compasscommand for all on join in eventCommands.yml
like `cmi compass world 1000 -2000 [playerName] -s`  this would define the compass to show to location 1000X and -2000Z in world world to all players that login your server. Dont forget to let ShowCompassTarget on true.

here you see the description from wiki.
 ```
cmi [compass (targetName) (sourceName) (x) (z) (worldname) (reset) (-s)](https://www.zrips.net/cmi/commands/)
Set players compass point to your location

Example:

/cmi compass Zhax
/cmi compass Zrips Zhax
/cmi compass LT_Craft 0 0 Zrips -s
/cmi compass reset Zrips
```

**Comment by Zrips** [comment]
This was a bug with was fixed with latest versions.
